subsys/bluetooth/controller/util/mayfly.c

```diff
@@ -171,11 +167,6 @@ void mayfly_run(u8_t callee_id)
 	u8_t enable = 0U;
 	u8_t caller_id;
 
-	if (!mfp[callee_id]) {
-		return;
-	}
-	mfp[callee_id] = 1U;
```

Comment:
You have in fact found a bug. Here it should be assigned to 0U.

This flag would allow mayfly_run to be placed in shared IRQ vector, and would only run through the loop if the pending was set.

Example, when LLL and ULL are configure to be at equal IRQ priority level, then only one (say, Radio) IRQ can be used for LLL and ULL context. When only that (Radio) IRQ asserts, mayfly_run would short circuit if this flag is not set.

That said, current upstream implementation does not have the code to share Radio ISR for ULL and LLL in single ISR.

Use of this implementation will reduce the number of IRQ vector entries used by the controller when sharing ISRs.

Action: Fix the assignment to be  0U. And enclose the mfp code in a new Kconfig which defaults to 'y' when LLL_PRIO == ULL_HIGH_PRIO OR ULL_HIGH_PRIO == ULL_LOW_PRIO.

Reply:
Hi Vinayah,

Sorry for the delay in reply; I've had to reconfigure my work environment and also a small vacation.
I don't really understand the reasoning of your reply. The function ```mayfly_run(u8_t callee_id)``` is executed for each callee (which corresponds to different IRQs). So if you have 2 IRQs calling the same function you would still affect different values of the mfp[] array:
```
IRQ0 -> mayfly_run(0) -> mfp[0]
IRQ1 -> mayfly_run(1) -> mfp[1]
```

My understanding was that ```mfp[]``` array was a shorter way of exiting ```mayfly_run()``` when there was no pending work to be done. But what I've stated in the commit message is that the implementation is wrong and at the same time eliminating the code doesn't lead to errors since  the code will just test that each list in the ```mft[callee][]``` array is empty and then exit.

If my understanding is correct the only reason for ```mfp[]``` would be improved efficiency.

Reply:
I mean:
Assume Counter compare IRQ is CNTR_IRQn and Software IRQ is SWI_IRQn, then
``
CNTR_IRQ -> Shared_ISR -> ticker_trigger() -> mayfly_run(0) -> mfp[0]
``
``
mayfly_enqueue -> SWI_IRQ -> Shared_ISR -> ticker_trigger() -> mayfly_run(0) -> mfp[0]
``

In the above either or both of ticker_trigger and mayfly_run(0) will check and decide to execute their respective code. In the case of mayfly_run, it will be the mfp[0] flag check.

Reply:
So, if ```CNTR_IRQ``` runs ```mayfly_run(0)``` with equal priority as ```mayfly_enqueue()``` this means it will not be preempted and therefore will execute all the queued jobs. Then when ```mayfly_enqueue()``` runs it will set mfp[0] = 1 and pend SWI_IRQ and therefor mayfly_run(0) will execute again.
So what is different?